clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1023,7 +1023,7 @@ static void checkExprLifetimeImpl(Sema &SemaRef,
         return false;
       }
 
-      if (IsGslPtrInitWithGslTempOwner && DiagLoc.isValid()) {
+      if (InitEntity && IsGslPtrInitWithGslTempOwner && DiagLoc.isValid()) {
```

Comment:
I think it would be simpler to just handle `AEntity` before:
```cpp
if(AEntity) {
  assert(shouldLifetimeExtendThroughPath(Path) == PathLifetimeKind::NoExtend);
  assert(LK == LK_Extended);
  if (!pathContainsInit(path)) {
    SemaRef.Diag(DiagLoc, diag::warn_dangling_pointer_assignment)
              << AEntity->LHS << DiagRange;
  }
  return;
}
assert(InitEntity);
switch (LK) {
...
<remove assert(InitEntity*) checks>
```

Reply:
Done.